lib/driver.md

```diff
@@ -0,0 +1,374 @@
+```
+
+`kore-exec`
+-----------
+
+```k
+    syntax PrePattern ::= koreExec(config: PrePattern) [seqstrict(1), result(Pattern)]
+                        | koreExec(file:   PreString)  [seqstrict(1), result(String)]
+    rule koreExec(Configuration) => koreExec(writeTempFile(unparsePattern(Configuration)))
+    rule koreExec(File)
+      => parse( system("kore-exec " +String michelsonDefinition() +String "/michelson-kompiled/definition.kore" +String
+                           " --strategy all" +String
+                           " --module MICHELSON" +String
+                           " --pattern " +String File
+                      )
+              )
+```
+
+Utilities instatiated for KMichelson
+------------------------------------
+
+```k
+    syntax  KoreName ::= "Lbl'-LT-'k'-GT-'" [token]
+    syntax Patterns ::= getKCell(Pattern) [function]
+    rule getKCell(Term) => findSubTermsByConstructor(Lbl'-LT-'k'-GT-', Term)
+
+    syntax  KoreName ::= "Lbl'-LT-'returncode'-GT-'" [token]
+    syntax Patterns ::= getReturncodeCell(Pattern) [function]
+    rule getReturncodeCell(Term) => findSubTermsByConstructor(Lbl'-LT-'returncode'-GT-', Term)
+
+    syntax PrePattern ::= parse(input: PreString)
+    rule parse(Input) =>  parse(Input, driverDefiniton() +String "/driver-kompiled/parser_Pattern")
+
+    syntax String ::= michelsonDefinition() [function, functional]
+    rule [[ michelsonDefinition() => DefnDir +String "/symbolic/" ]]
+         <defnDir> DefnDir </defnDir>
+
+    syntax String ::= driverDefiniton() [function, functional]
+    rule [[ driverDefiniton() => DefnDir +String "/driver/" ]]
+         <defnDir> DefnDir </defnDir>
+```
+
+
+Main
+----
+
+We initialize the configuration with the input filename.
+
+```k
+    configuration <k> init($InputFilename) </k>
+                  <success>  0 </success>               // Number of successful branches
+                  <failures> 0 </failures>              // Number of failing    branches
+                  <defnDir> $DefnDir:String </defnDir>  // Path to kompiled definitions
+                  <out stream="stdout"> .List </out>
+                  <exitcode exit="0"> 2 </exitcode>
+```
+
+
+### Initialization
+
+We then parse the program syntax into kore using `llvm-krun`, and then parse the kore string into a `Pattern` we then exuecute this pattern with `kore-exec`.
+
+```k
+    syntax KItem ::= init(filename: String)
+    rule  <k> init(ProgramFile)
+           => koreExec(initialConfiguration(ProgramFile, michelsonDefinition() +String "/michelson-kompiled/"))
+              ...
+          </k>
+
+    syntax PrePattern ::= initialConfiguration(filename: PreString, definition: String) [seqstrict(1), result(String)]
+    // TODO: The parser here isn't generated
+    rule initialConfiguration(Filename, Definition)
+      => parse( system( "llvm-krun --dry-run --directory " +String Definition
+                        +String " -c PGM " +String Filename +String " Pgm prettyfile") 
```

Comment:
There may be a more modern way of achieving this. There should be a list of all K binaries that are useful somewhere...